sbt-plugin/src/main/scala-sbt-1.0/org/scalajs/sbtplugin/SBTCompat.scala

```diff
@@ -0,0 +1,68 @@
+package org.scalajs.sbtplugin
+
+import sbt._
+
+import org.scalajs.core.tools.io.FileVirtualFile
+
+private[sbtplugin] object SBTCompat {
+  type IncOptions = xsbti.compile.IncOptions
+
+  type CompileAnalysis = xsbti.compile.CompileAnalysis
+
+  val formatImplicits: sjsonnew.BasicJsonProtocol.type =
+    sjsonnew.BasicJsonProtocol
+
+  def moduleIDWithConfigurations(moduleID: ModuleID,
+      configurations: Option[String]): ModuleID = {
+    moduleID.withConfigurations(configurations)
+  }
+
+  def crossVersionAddScalaJSPart(cross: CrossVersion,
+      part: String): CrossVersion = {
+    cross match {
+      case CrossVersion.Disabled =>
+        CrossVersion.constant(part)
+      case cross: sbt.librarymanagement.Constant =>
+        cross.withValue(part + "_" + cross.value)
+      case cross: CrossVersion.Binary =>
+        cross.withPrefix(part + "_" + cross.prefix)
+      case cross: CrossVersion.Full =>
+        cross.withPrefix(part + "_" + cross.prefix)
```

Comment:
For reference, this is the definition of `CrossVersion.Binary`: [`Binary.scala`](https://github.com/sbt/librarymanagement/blob/60dfb6e0b1a1220256f866a55ccda09c16344f4e/core/src/main/contraband-scala/sbt/librarymanagement/Binary.scala). Other `CrossVersion`s are similar.

They are similar to our binary-compatible-forever `Config` objects in shape, except in sbt they auto-generate them from [a big JSON file](https://github.com/sbt/librarymanagement/blob/60dfb6e0b1a1220256f866a55ccda09c16344f4e/core/src/main/contraband/librarymanagement.json#L253), using [contraband](http://www.scala-sbt.org/contraband/).